src/k8s-engine/wsl.ts

```diff
@@ -873,6 +875,41 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     const nerdctlPath = await this.wslify(windowsNerdctlPath);
 
     await this.runInstallScript(INSTALL_WSL_HELPERS_SCRIPT, 'install-wsl-helpers', nerdctlPath);
+    await this.installCredentialHelper();
+  }
+
+  protected async getHostIPAddr(): Promise<string> {
+    try {
+      const lines = (await this.execCommand({ capture: true }, '/sbin/ip', 'route', 'list', 'eth0')).split(/\n/);
+      const fields = lines[0].split(/\s+/);
+
+      return fields[2];
+    } catch (err: any) {
+      console.log(`ip route failed: ${ err }`, err);
+      throw err;
+    }
+  }
+
+  protected async installCredentialHelper() {
+    const credsPath = getServerCredentialsPath();
+
+    try {
+      const hostIPAddr = await this.getHostIPAddr();
+      const stateInfo: ServerState = JSON.parse(await fs.promises.readFile(credsPath, { encoding: 'utf-8' }));
+      const escapedPassword = stateInfo.password.replace(/\\/g, '\\\\')
+        .replace(/"/g, '\\"');
+      const fileContents = `CREDFWD_AUTH="${ stateInfo.user }:${ escapedPassword }"
+CREDFWD_URL="http://${ hostIPAddr }:${ stateInfo.port }"
```

Comment:
See the lima version — need to use single quote or escape dollar.

Reply:
Yes